src/main/java/com/cloudbees/jenkins/support/impl/RunningJobs.java

```diff
@@ -0,0 +1,53 @@
+package com.cloudbees.jenkins.support.impl;
+
+import com.cloudbees.jenkins.support.api.Component;
+import com.cloudbees.jenkins.support.api.Container;
+import com.cloudbees.jenkins.support.api.PrefilteredPrintedContent;
+import com.cloudbees.jenkins.support.filter.ContentFilter;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.security.Permission;
+import jenkins.model.Jenkins;
+import org.jenkinsci.plugins.workflow.support.steps.ExecutorStepExecution;
+
+import java.io.PrintWriter;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Extension
+public class RunningJobs extends Component {
+
+    @NonNull
+    @Override
+    public Set<Permission> getRequiredPermissions() {
+        return Collections.singleton(Jenkins.ADMINISTER);
+    }
+
+    @Override
+    @NonNull
+    public String getDisplayName() {
+        return "Running Jobs";
+    }
+
+    @Override
+    public void addContents(@NonNull Container result) {
+        result.add(
+            new PrefilteredPrintedContent("nodes/master/running-jobs.txt") {
```

Comment:
Perhaps this file should be at top level? It would be pretty misleading for something under `nodes/master/` to list freestyle builds running on an agent. Those _could_ be moved to a separate `nodes/*/running-builds.txt`, though I am not sure if that is helpful or just annoying.

In the case of Pipeline builds, there is no unique associated node, though in many cases there will be exactly one `PlaceholderTask`. (There could be several, if in `parallel`, or none, if paused for example in `input`.)

Reply:
yes, that makes sense.